ads/google/doubleclick.js

```diff
@@ -87,6 +117,80 @@ function doubleClickWithGpt(global, data, gladeExperiment) {
   container.style.right = '';
   container.style.transform = 'translate(-50%, -50%)';
 
+  // Get multi-size ad request data, if any, and validate it in the following
+  // ways: Ensure that the data string is a comma-separated list of sizes of the
+  // form wxh; that each secondary dimension is strictly less than its primary
+  // dimension counterpart; and, if data-mutli-size-validation is not set to
+  // false, that each secondary dimension is at least 2/3rds of its primary
+  // dimension counterpart.
+  const multiSizeDataStr = data.multiSize || null;
+  if (multiSizeDataStr) {
+    const sizesStr = multiSizeDataStr.split(',');
+    sizesStr.forEach(sizeStr => {
+
+      const size = sizeStr.split('x');
+
+      // Make sure that each size is specified in the form val1xval2.
+      if (!Array.isArray(size) || size.length != 2) {
+        user().error(TAG_, 'Invalid multi-size data format.', '<amp-ad>');
+        return;
+      }
+
+      const widthStr = size[0];
+      const heightStr = size[1];
+
+      // Make sure that both dimensions given are numbers.
+      let badParams = getBadParams(widthStr, heightStr,
```

Comment:
I'm hesitating on this feedback, but how about:

```
if (!validateDimensions(width, height, 
    w => {...}, 
    h => {...},
    badParams => errorMsg)) {
  return;
}
```
where `validateDimensions` returns a boolean.

I just feel that the function `getBadParams` is less intuitive.